pkg/controller/machinepool/actuator.go

```diff
@@ -22,9 +22,17 @@ type Actuator interface {
 	// to wait before we can proceed with reconciling. (e.g. obtaining a pool name lease)
 	GenerateMachineSets(*hivev1.ClusterDeployment, *hivev1.MachinePool, log.FieldLogger) (msets []*machineapi.MachineSet, proceed bool, genError error)
 
-	// GenerateCAPIMachineSets generates upstream CAPI MachineSets
+	// GenerateCAPIMachineSets returns the desired set of CAPI MachineSets in the target namespace on the hive cluster
+	// for a given MachinePool.
+	// Returns the list of generated machine sets, a boolean indicating if the controller should proceed with reconcile
+	// or not, and an error. The boolean may be set in situations where we have not encountered an error, but still need
+	// to wait before we can proceed with reconciling. (e.g. obtaining a pool name lease)
 	GenerateCAPIMachineSets(*hivev1.ClusterDeployment, *hivev1.MachinePool, log.FieldLogger) (msets []*capiv1.MachineSet, proceed bool, genError error)
 
-	// GenerateMachineTemplates()
+	// GenerateMachineTemplates returns the desired set MachineTemplates for CAPI MachineSets in the target namespace on
+	// the hive cluster for a given MachinePool.
+	// Returns the list of generated machine sets, a boolean indicating if the controller should proceed with reconcile
+	// or not, and an error. The boolean may be set in situations where we have not encountered an error, but still need
+	// to wait before we can proceed with reconciling. (e.g. obtaining a pool name lease)
 	GenerateMachineTemplates(*hivev1.ClusterDeployment, *hivev1.MachinePool, log.FieldLogger) (templates interface{}, proceed bool, genError error)
```

Comment:
I think we shuold atleast replace the template interface {} with templates []client.Object